src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py

```diff
@@ -423,32 +399,45 @@ def _generate_pre_test_check_command(self, cmd_args: Dict[str, str], output_path
 
         return "\n".join(
             [
-                f'pretest_output_files="{pretest_output_files}"',
+                f'PRETEST_OUTPUT_FILES="{pretest_output_files}"',
                 f'keyword="{keyword}"',
                 "",
                 "# Use grep to search for the keyword in the files",
-                'if grep -q "$keyword" $pretest_output_files; then',
-                "    keyword_found=true",
+                'if grep -q "$keyword" $PRETEST_OUTPUT_FILES; then',
+                "    PRE_TEST_SUCCESS=true",
                 "fi",
             ]
         )
 
-    def _generate_container_load_srun_command(self, container_image: str) -> str:
-        """
-        Generate an srun command to load a container using the specified image.
+    def _generate_container_load_command(self, slurm_args: Dict[str, Any]) -> List[str]:
+        """Generate the command for loading a container."""
+        commands = []
+        container_image = slurm_args.get("image_path")
+        if not container_image:
+            raise ValueError("image_path in slurm_args must be a valid path")
 
-        Args:
-            container_image (str): The path to the container image.
+        commands.append('if [ "$PRE_TEST_SUCCESS" = true ]; then')
```

Comment:
Let's avoid doing multiple function call and use list creation as `l = [...]`, same for the code below.

Reply:
Sorry, I wasn't clear. Let's change 
```py
    ...
    commands.append('    echo "Running srun command"')
    commands.append("    srun \\")
    commands.append("    --mpi=none \\")
    ...
```
to
```py
cmds = [
    ...
    '    echo "Running srun command"',
    "    srun \\",
    "    --mpi=none \\",
    ...
]
```

Reply:
Thanks, updated.